### PR TITLE
my small contirbution (in particular the dont dos ;))

### PR DESCRIPTION
**Pull request by rmannibucau**
Since we exchanged on twitter about that here is my version ;)

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Hey @rmannibucau, could you resolve those merge conflicts, rebase to latest main, and squash everything into one commit? Also please make sure that `./test.sh rmannibucau` passes. Thx!

**Comment by rmannibucau** [comment]
hi @gunnarmorling fixed it normally

side note: got some license header issues for some other classes, didn't fix it to limit the diff

**Comment by gunnarmorling** [comment]
> side note: got some license header issues for some other classes, didn't fix it to limit the diff

Interesting. Any actual build failure due to that? Can't seem to reproduce this as of current main.

File: src/main/java/dev/morling/onebrc/CalculateAverage_rmannibucau.java [hunk 1]
```diff
@@ -0,0 +1,736 @@
...
+
+    // NOTE: this is a work code which shouldn't stay but kept to explain the rational behind some impl choice (in Bytes) for the exercise
+    // we assume the station set is closed as of today and optimize Bytes equals impl by checking conflicts only per hash
+    public static class ComputePotentialStationDiffsPerHash {
+        private ComputePotentialStationDiffsPerHash() {
+            // no-op
+        }
+
+        public static void main(final String... args) {
+            final var stations = List.of("Abha",
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/onebrc/CalculateAverage_rmannibucau.java:304]: Hum, there should be no assumptions about the specific data set (see rules). Not quite clear to me what's happening here?

File: calculate_average_rmannibucau.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+# $script build -> build on 21-zulu before running
+# $script native -> use native-image and build before running
+
+function run() {
+  JAVA_OPTS="--enable-preview"
+  java $JAVA_OPTS --class-path target/classes dev.morling.onebrc.CalculateAverage_rmannibucau
+}
+
+if [ -n "$1" ]; then
```
Comment (gunnarmorling) [inline calculate_average_rmannibucau.sh:28]: So do you want to submit this as native or JVM? If native, please extract the build step into a separate script. See Thomas Würthinger's implementation for an example.

**Review by gunnarmorling — CHANGES_REQUESTED** [review]

**Comment by rmannibucau** [comment]
@gunnarmorling got fixed with some rebasing, guess was due to some diff and fixed since, forget it (the license thing)

File: src/main/java/dev/morling/onebrc/CalculateAverage_rmannibucau.java (hunk 1 above)
Comment (rmannibucau) [reply src/main/java/dev/morling/onebrc/CalculateAverage_rmannibucau.java:304]: well, part of what I'm interesting in there is the extreme abuses of challenges on one side but also the fact that some choices don't work better than default impl (linear hash based structures here).
This main (which is totally useless for the runtime, as written, is really to explain why the impl works) is intended to ensure the hashCode distributes the (pseudo-expected) values properly for a hashmap.
Concretely the idea was to validate "offline" Bytes#equals implementation assuming the names are (pseudo)open but compatible with such an implementation - which is generally the case with such input data.

side note: this was also validated with https://github.com/meteostat/weather-stations (21K+ station names) and even if conflicts are possible they are unlikely enough - or would probably be handled in the file generator (depends a bit the ecosystem around) so for me it is not a so unfair assumption.

That said I can totally understand you think it does not respect the rule, no big deal to not integrate the PR for me.
Don't think I would be in the top results anyway in 20 days with this code and just used holidays time to participate ;).

File: calculate_average_rmannibucau.sh (hunk 2 above)
Comment (rmannibucau) [reply calculate_average_rmannibucau.sh:28]: jvm, graalvm boost is not impressive enough to be relevant IMHO - it is literally in the error zone - (was some ms on my tests/computer, just kept it in the script to enable it and potentially tune it more but default run is really what should be evaluated)

File: src/main/java/dev/morling/onebrc/CalculateAverage_rmannibucau.java (hunk 1 above)
Comment (gunnarmorling) [reply src/main/java/dev/morling/onebrc/CalculateAverage_rmannibucau.java:304]: Yeah, indeed I am inclined to close the PR in its current form, as there shouldn't be any specific asumptions or handling of the example station names from the data set, as per the rules. Thanks a lot for participating nevertheless!

**Comment by gunnarmorling** [comment]
Closing, as discussed.

**Pull request closed without merging**